pkg/resources/management.cattle.io/v3/fleetworkspace/mutator_test.go

```diff
@@ -0,0 +1,96 @@
+package fleetworkspace
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/webhook/pkg/admission"
+	"github.com/rancher/wrangler/pkg/generic/fake"
+	"github.com/stretchr/testify/require"
+	admissionv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const nsName = "test"
+
+func TestAdmit(t *testing.T) {
+	workspace := v3.FleetWorkspace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nsName,
+		},
+	}
+	raw, _ := json.Marshal(workspace)
+	req := &admission.Request{
+		AdmissionRequest: admissionv1.AdmissionRequest{
+			Object: runtime.RawExtension{Object: &workspace, Raw: raw},
+		},
+	}
+
+	tests := map[string]struct {
+		m                  func(t *testing.T) Mutator
+		req                *admission.Request
+		expectAllowed      bool
+		expectResultStatus *metav1.Status
+	}{
+		"reject because namespace already exists": {
```

Comment:
Let's add the following test cases to hit all of the changes made in this PR.

- namespace already exists with the expected label
- namespace already exists with the expected label, but the user fails the escalation check
- failed to create a namespace with an error other than alreadyExist.

To add tests that hit the escalation check, you will need to create a resolver for your mutator there is an example of this [here](https://github.com/rancher/webhook/blob/release/v0.4/pkg/resources/management.cattle.io/v3/roletemplate/validator_test.go#L34C1-L50)

Reply:
Done. Please check the latest commit.